pkg/storage/storage.go

```diff
@@ -142,6 +151,52 @@ func (s *Storage) History(name string) ([]*rspb.Release, error) {
 	return s.Driver.Query(map[string]string{"NAME": name, "OWNER": "TILLER"})
 }
 
+// removeLeastRecent removes items from history until the lengh number of releases
+// does not exceed max.
+//
+// We allow max to be set explicitly so that calling functions can "make space"
+// for the new records they are going to write.
+func (s *Storage) removeLeastRecent(name string, max int) error {
+	if max <= 0 {
+		return nil
+	}
+	h, err := s.History(name)
+	if err != nil {
+		s.Log("error fetching history: %s", err)
+		return err
+	}
+	if len(h) <= max {
+		return nil
+	}
+	overage := len(h) - max
+
+	// We want oldest to newest
+	relutil.SortByRevision(h)
+
+	// We want to try to delete as many as possible because some systems
```

Comment:
- [ ] The two occurrences of "because" confused me. A suggested rewording:  
  > Even in the face of errors due to API throughput limitations, we try to delete as many releases as possible now, assuming that we'll try again later to make gradual progress toward meeting the desired limit.